src/controllers/user.controller.js

```diff
@@ -0,0 +1,138 @@
+import { validationResult } from 'express-validator';
+import bcrypt from 'bcrypt';
+import { v4 as uuidv4 } from 'uuid';
+
+import * as userService from '../services/user.service.js';
+import * as emailService from '../services/email.service.js';
+import * as tokenService from '../services/token.service.js';
+
+import ApiError from '../exeptions/api.error.js';
+import { hashPassword } from '../utils/helpers.js';
+
+export const changeName = async (req, res) => {
+  const validationResults = validationResult(req);
+
+  if (!validationResults.isEmpty()) {
+    throw ApiError.badRequest(validationResults.array());
+  }
+
+  const { name } = req.body;
+
+  const { email } = res.locals.user;
+
+  const user = await userService.findUser(email);
+
+  if (!user) {
+    throw ApiError.notFound('User not found');
+  }
+
+  user.name = name;
+  await user.save();
```

Comment:
All logic related to interaction with the database should be placed in repositories.
You can check articles about repository pattern, it helps to simplify interaction with a database, so when you change it or ORM you can easily migrate
https://4markdown.com/understanding-repository-pattern-in-nodejs-and-typescript/